shared/js/background/https.es6.js

```diff
@@ -99,40 +99,19 @@ class HTTPS {
 
                 chrome.storage.local.get('https-upgrade-list', (results) => {
                     if (!results || !results['https-upgrade-list']) {
-                        return reject()
+                        return reject(new Error('couldn\'t get HTTPS results from local storage'))
                     }
 
                     try {
                         let parsedList = JSON.parse(results['https-upgrade-list'])
                         resolve(parsedList)
                     } catch (e) {
                         console.log('HTTPS: getFromStorage() error parsing JSON from chrome.storage.local', e)
-                        reject()
+                        reject(e)
                     }
                 })
-
-            // For Safari
-            } else if (window.localStorage) {
-                console.log('HTTPS: getFromStorage() using localStorage (Safari)')
-
-                if (!window.localStorage || !localStorage['https-upgrade-list0']) {
-                    return reject()
-                }
-
-                let data = ''
-                for (let i = 0; i < LOCAL_STORAGE_CHUNKS; i++) {
-                    data += localStorage['https-upgrade-list' + i]
-                }
-
-                try {
-                    let parsedList = JSON.parse(data)
-                    resolve(parsedList)
-                } catch (e) {
-                    console.log('HTTPS: getFromStorage() error parsing JSON from localStorage', e)
-                    reject()
-                }
```

Comment:
This was a bit of a weird one.

`LOCAL_STORAGE_CHUNKS` wasn't defined anywhere. When this runs (if it runs?) the for statement would evaluate to `0 < undefined` which returns false. So we'd always end up with a rejection anyway.

We're not using HTTPS in Safari anyway, so I just removed this.